packages/services/src/kernel/default.ts

```diff
@@ -1375,26 +1354,21 @@ export class KernelConnection implements Kernel.IKernelConnection {
           // Updating the status is synchronous, and we call no async user code
           const executionState = (msg as KernelMessage.IStatusMsg).content
             .execution_state;
-          if (executionState === 'autorestarting') {
+          if (executionState === 'restarting') {
             // The kernel has been auto-restarted by the server. After
             // processing for this message is completely done, we want to
             // handle this restart, so we don't await, but instead schedule
             // the work as a microtask (i.e., in a promise resolution). We
             // schedule this here so that it comes before any microtasks that
             // might be scheduled in the status signal emission below.
             void Promise.resolve().then(async () => {
-              // handleRestart changes the status to 'restarting', so we call it
-              // first so that the status won't flip back and forth between
-              // 'restarting' and 'autorestarting'.
-              await this._handleRestart();
               this._updateStatus('autorestarting');
-            });
-          }
-          if (executionState === 'restarting') {
-            void Promise.resolve().then(async () => {
-              await this._handleRestart();
-              this._kernelSession = msg.header.session;
-              this._updateStatus('restarting');
+              this._clearKernelState();
```

Comment:
Instead of calling out to _handleRestart, we just incorporate the logic here since this was the only place calling the method. As with other cases, we change the status before clearing the kernel state. We don't need to set the _kernelSession variable since the kernel restart already happened, so we don't need to set it to the sentinel, and the _kernelSession is being reset to the id sent from the server.